lib/internal/freelist.js

```diff
@@ -1,24 +1,25 @@
 'use strict';
 
-// This is a free list to avoid creating so many of the same object.
-exports.FreeList = function(name, max, constructor) {
-  this.name = name;
-  this.constructor = constructor;
-  this.max = max;
-  this.list = [];
-};
-
-
-exports.FreeList.prototype.alloc = function() {
-  return this.list.length ? this.list.pop() :
-                            this.constructor.apply(this, arguments);
-};
+class FreeList {
+  constructor(name, max, constructor) {
+    this.name = name;
+    this.constructor = constructor;
+    this.max = max;
+    this.list = [];
+  }
 
+  alloc() {
+    return this.list.length ? this.list.pop() :
+                              this.constructor.apply(this, arguments);
```

Comment:
Worth playing around with. I would also note that I had attempted to replace this use of `arguments` with `...args`, but found that in the case there are no arguments passed to the `alloc()`, it caused a whopping 48% slowdown. Definitely something we'll need to keep in mind.